cli/internal/run/graph_run.go

```diff
@@ -0,0 +1,46 @@
+package run
+
+import (
+	gocontext "context"
+
+	"github.com/pyr-sh/dag"
+	"github.com/vercel/turbo/cli/internal/cmdutil"
+	"github.com/vercel/turbo/cli/internal/core"
+	"github.com/vercel/turbo/cli/internal/graphvisualizer"
+	"github.com/vercel/turbo/cli/internal/util"
+)
+
+// GraphRun generates a visualization of the task graph rather than executing it.
+func GraphRun(ctx gocontext.Context, rs *runSpec, engine *core.Engine, base *cmdutil.CmdBase) error {
```

Comment:
I think we should make this private unless we expect this to be used outside of `run`.

(Bikeshed) This name confused me for a second thinking it was running a graph. Maybe `renderGraph` or `generateGraph`?

Reply:
I wanted it to be separate packages in the beginning, but ultimately decided against it. Does private/public make a big difference in this case? I wanted to model the three different execution types that used to be like this:

```golang
if graph {
	// generate graph
} else if dryRun {
	// do the dry run
} else {
	// execute
}
```

and turn each of those branches into `GraphRun`, `DryRun`, and `RealRun`. Open to bikeshed, but I wanted the naming pattern to make it obvious that they are the three things that can happen with `turbo run`. 

Down the road, expect that they they won't be mutually exclusive (e.g. Run Summary is _basically_ half implemented as DryRun already, and I don't see why we can't generate a TaskGraph image even on a real runs to describe what happened (similar to `--profile`). 

Does this context change your mind or help see where I'm coming from?

Reply:
That context helps. IMO the `run` package does a good enough job of conveying that these functions happen during `turbo run` and we can drop the `Run` from from `GraphRun` and `RealRun` (`DryRun` reads correctly to me). Just my 2¢

Reply:
Yeah, maybe it's just a stylistic issue then at the moment. Another thing I was hoping to do (but won't spend any more time on right now) was to find a good abstraction for is currently `run{} + execContext{}` structs, so they're more general applicable. E.g. `execContext.exec()` is inly used in `RealRun()`, but with some changes, it seems like it would be helpful for both other types of runs also.

Another thing I'm thinking about is that if we can clean up which logic is used where, we could, for example free up `DryRun()` and `GraphRun()` for Rust-ports more easily, since it'll leave the boundaries pretty clean. 

But fully agree this PR doesn't go far enough for any of that, it really just breaks up the `run.go` file... which doesn't feel as compelling.